test/cmd/images_tests.sh

```diff
@@ -88,6 +88,30 @@ os::cmd::expect_success_and_not_text 'oc get istag/wildfly:latest -o jsonpath={.
 os::cmd::expect_success "oc patch istag/wildfly:latest -p='{\"tag\":{\"importPolicy\":{\"scheduled\":true}}}'"
 os::cmd::expect_success_and_text 'oc get istag/wildfly:latest -o jsonpath={.tag.importPolicy.scheduled}' 'true'
 
+# create an image stream tag
+os::cmd::expect_success 'oc create imagestreamtag tag:1 --from=wildfly:latest'
+os::cmd::expect_success 'oc create imagestreamtag tag:2 --from-image=mysql:latest'
+os::cmd::try_until_success 'oc get imagestreamtags tag:2'
+os::cmd::expect_success 'oc create imagestreamtag tag:3 -A foo=bar'
+os::cmd::expect_success 'oc create imagestreamtag tag:4 --from=:2'
+os::cmd::expect_success 'oc create imagestreamtag tag:5 --from=tag:2'
+os::cmd::expect_success 'oc create imagestreamtag tag:6 --reference --from-image=mysql:latest'
+os::cmd::expect_success 'oc create imagestreamtag tag:7 --reference-policy=Local --from=tag:2'
+os::cmd::expect_success 'oc create istag tag:8 --insecure --from-image=mysql:latest'
+os::cmd::try_until_success 'oc get imagestreamtags tag:8'
+os::cmd::expect_success 'oc create imagestreamtag tag:9 --scheduled --reference-policy=Local --from-image=mysql:latest'
+
+os::cmd::expect_success_and_text 'oc get istag/tag:2 -o jsonpath={.image.dockerImageReference}' 'mysql@sha256:'
+tag2=$( oc get istag/tag:2 -o jsonpath={.image.metadata.name} )
+os::cmd::expect_success_and_text "oc get is/tag -o 'jsonpath={.spec.tags[?(@.name==\"3\")].annotations.foo}'" 'bar'
+os::cmd::expect_success_and_text 'oc get istag/tag:4 -o jsonpath={.image.metadata.name}' "${tag2}"
+os::cmd::expect_success_and_text "oc get is/tag -o 'jsonpath={.spec.tags[?(@.name==\"4\")].from.name}'" '^2$'
+os::cmd::expect_success_and_text 'oc get istag/tag:5 -o jsonpath={.image.metadata.name}' "${tag2}"
+os::cmd::expect_success_and_text "oc get is/tag -o 'jsonpath={.spec.tags[?(@.name==\"6\")].reference}'" 'true'
+os::cmd::expect_success_and_text "oc get is/tag -o 'jsonpath={.spec.tags[?(@.name==\"7\")].referencePolicy}'" 'Local'
+os::cmd::expect_success_and_text "oc get is/tag -o 'jsonpath={.spec.tags[?(@.name==\"8\")].importPolicy.insecure}'" 'true'
+os::cmd::expect_success_and_text "oc get is/tag -o 'jsonpath={.spec.tags[?(@.name==\"9\")].importPolicy.scheduled}'" 'true'
+
```

Comment:
I'm missing following tests:
1. Create tag with default tag name (`latest`) applied automatically (iow. `oc create istag mytag`) since I don't see any requirement on tag name being in the form `repo:tag`. 
2. Create tag inside an existing IS, a `oc create is` + `oc create istag` sequence would be nice. 
3. Negative tests (cli validation faliures, creating already existing tag, etc).

Reply:
We don't do implicit latest here, adding a negative test.  Will test inside.